doc/pages/development-guide/001-dev_guide.md

```diff
@@ -0,0 +1,165 @@
+
+- [Astarte Core](https://github.com/astarte-platform/astarte_core): contains definitions common to the application, e.g. Realm name, Device ID validity, Interface definitions etc.
+- [Astarte RPC](https://github.com/astarte-platform/astarte_rpc): contains all the Protobufs and code used in the internal RPC mechanism. In most cases, RPCs are between an API component and the related backend.
+- [Astarte Data Access](https://github.com/astarte-platform/astarte_data_access): contains some interactions with the database that are common to all Astarte services, e.g. queries for interface values.
+
+### Documentation
+
+Documentation is automatically generated from files in the `doc/pages` subdirectory of the [main Astarte repository](https://www.github.com/astarte-platform/astarte) on Github.
+It is written in Github-flavored markdown.
+
+### Tooling
+
+The main Astarte repo contains a number of tools that can be used everyday when interacting with Astarte. They are found in the `tools` directory subdirectory of the [main Astarte repository](https://www.github.com/astarte-platform/astarte) on Github:
+
+- Astarte Device Fleet Simulator: create a fleet of virtual devices. Mainly used for load testing
+- Astarte E2E: a tool that sends data from a virtual device and checks the resulting value in Astarte using triggers. Mainly used for monitoring
+- Astarte Import and Astarte Export: used to load/export data into/from Astarte
+
+## Can I use an Elixir REPL to help me during development?
+
+Sure thing! However, usually Astarte services do need some context around them.
+Refer to the following [”Testing a single component”](#testing-a-single-component) paragraph for dependencies, then you can start your application with an interactive shell by replacing `mix test` with `iex -S mix`
+
+## Ok, I made changes, what now?
+
+Once you got your hands dirty, is time to test them.
+You can test changes to a single component, or run E2E tests if more than one service is involved.
+
+Remember that a contribution should always include tests for the new functionality or the fix.
+We use the [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) framework for unit testing.
+Work is under way to add property-based testing too! 
+
+### Testing a single component
+
+In general, you will have to bring up a RabbitMQ and a Scylla instance (not all services need both: API services will not need access to the database).
+You can do so by running
+
+```bash
+docker run --rm -p 5672:5672 -p 15672:15672 rabbitmq:management
+docker run --rm -p 9042:9042 scylladb/scylla
+```
+
+Then, you can run test in the component directory with
+
+```bash
+RABBITMQ_HOST=localhost CASSANDRA_NODES=localhost mix test --exclude wip
+```
+
+> [!NOTE]
+> Some services are a bit special (for now!) and might need a little more setup for testing.
+> - AppEngine API needs the AMQP exchange `astarte_events` to be declared:
+> after having started RabbitMQ, run the following lines to declare it:
+>  ```bash
```

Comment:
while it's good to give users something to copy and ready to use, we're talking to download and make executable a file from a third party, then delete id. It's probably better to add some context about what is rabbitmqadmin and what could be done with it (link to official docs will suffice)

Reply:
On the other hand, if rabbitmq is started in detached mode and its stdout is assigned to a variable, we can just docker exec into it without needing use wget

```bash
RMQ_CONTAINER=$(docker run -d --rm -p 5672:5672 -p 15672:15672 rabbitmq:management)

docker exec $RMQ_CONTAINER rabbitmqadmin declare exchange name=astarte_events type=direct
```